TFNetworkRecLayer.py

```diff
@@ -4177,3 +4177,141 @@ def call(self, *args, **kwargs):
     return y
 
 
+class LayerNormLSTMCell(BaseRNNCell):
+  """LSTM unit with layer normalization and recurrent dropout
+
+  Layer normalization implementation is based on:
+        https://arxiv.org/abs/1607.06450
+
+  Recurrent dropout l2p
+  is based on:
+        https://arxiv.org/abs/1603.05118
+  """
+
+  def __init__(self,
+               num_units,
+               norm_gain=1.0,
+               norm_shift=0.0,
+               activation=tf.tanh,
+               dropout_keep_prob=1.0,
+               dropout_prob_seed=None,
+               with_bias=False,
+               with_concat=True,
+               global_norm=True,
+               per_gate_norm=False,
+               cell_norm=False,
+               hidden_norm=False):
+    """
+
+    :param num_units: An integer, the number of lstm units
+    :param norm_gain: A float, the layer normalization gain value
+    :param norm_shift: A float, the layer normalization shift (bias) value
+    :param activation: Activation function to be applied in the lstm cell
+    :param with_bias: A boolean, if True then bias is added to the linear
+      computation output
+    :param with_concat: A boolean, if True then the input and prev hidden state
+      is concatenated for the computation
+    :param global_norm: A boolean, if True then layer normalization is applied
+      for the forward and recurrent outputs
+    :param per_gate_norm: A boolean, if True then layer normalization is applied
+      per lstm gate
+    :param cell_norm: A boolean, if True then layer normalization is applied
+      to the LSTM new cell output
+    :param hidden_norm: A boolean, if True then layer normalization is applied
```

Comment:
Write the type (`int`, `float`, `bool`) in the common PEP/PyCharm format if possible (e.g. `:param int num_inits: number of LSTM units`).